src/application.hpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2019-2021 Ashar Khan <ashar786khan@gmail.com>
+ *
+ * This file is part of CP Editor.
+ *
+ * CP Editor is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * I will not be responsible if CP Editor behaves in unexpected way and
+ * causes your ratings to go down and or lose any important contest.
+ *
+ * Believe Software is "Software" and it isn't immune to bugs.
+ *
+ */
+
+#ifndef APPLICATION_HPP
+#define APPLICATION_HPP
+
+#include <QFileOpenEvent>
+#include <singleapplication.h>
+
+class Application : public SingleApplication
+{
+  public:
+    Application(int &argc, char **argv) : SingleApplication(argc, argv, true)
```

Comment:
Use explicit before ctor. Move the implementation to cpp file.

Reply:
explicit is required for constructors taking only 1 non default parameter

See:  https://en.cppreference.com/w/cpp/language/explicit

Reply:
that is until c++11. Its always good practice to have explicit before ctor

Reply:
If static analysis passed it's okay.